### PR TITLE
Modify fatalIf, startup and update message logging code

### PR DESCRIPTION
**Pull request by kannappanr**
## Description
Change fatalIf implementation in logger.go. Do not log trace, as all
of the fatalIf calls are made during startup.

Use new functions to and a common logging framework to make sure that
startup and update messages get logged in both console and json mode.

## Motivation and Context
These changes are part of error logging pod.

## How Has This Been Tested?
Manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/logger.go [hunk 1]
```diff
@@ -219,5 +225,89 @@ func errorIf(err error, msg string, data ...interface{}) {
...
+		Time:    UTCNow().Format(time.RFC3339Nano),
+	})
+	if err != nil {
+		panic("json marshal of logEntry failed: " + err.Error())
+	}
+	fmt.Println(string(logJSON))
+}
+
+func (i infoMsg) quiet(cause, msg string, args ...interface{}) {
+	i.pretty(cause, msg, args...)
```
Comment (ebozduman) [inline cmd/logger.go:283]: Don't we need to be quite for info messages when `--quite` flag is turned on?

File: cmd/prepare-storage.go [hunk 2]
```diff
@@ -113,7 +112,7 @@ func waitForFormatXL(firstDisk bool, endpoints EndpointList, setCount, disksPerS
 
 			if shouldInitXLDisks(sErrs) {
 				if !firstDisk {
-					console.Println("Waiting for the first server to format the disks.")
+					LogInfoMessage("Waiting for the first server to format the disks.\n")
```
Comment (ebozduman) [inline cmd/prepare-storage.go]: You already have a new line character, `\n`, added in `LogInfoMessage` definition.
Is there any  other reason you need it here again?

File: cmd/prepare-storage.go [hunk 3]
```diff
@@ -137,7 +136,7 @@ func waitForFormatXL(firstDisk bool, endpoints EndpointList, setCount, disksPerS
 				}
 				return format, nil
 			}
-			console.Printf("Waiting for a minimum of %d disks to come online (elapsed %s)\n", len(endpoints)/2, getElapsedTime())
+			LogInfoMessage("Waiting for a minimum of %d disks to come online (elapsed %s)\n", len(endpoints)/2, getElapsedTime())
```
Comment (ebozduman) [inline cmd/prepare-storage.go]: You already have a new line character, \n, added in LogInfoMessage definition.
Is there any other reason you need it here again?

File: cmd/signals.go [hunk 4]
```diff
@@ -62,14 +62,14 @@ func handleSignals() {
 			exit(err == nil && oerr == nil)
 		case osSignal := <-globalOSSignalCh:
 			stopHTTPTrace()
-			log.Printf("Exiting on signal %v\n", osSignal)
+			LogInfoMessage("Exiting on signal %v\n", osSignal)
```
Comment (ebozduman) [inline cmd/signals.go]: You already have a new line character, \n, added in LogInfoMessage definition.
Is there any other reason you need it here again?

**Review by ebozduman — CHANGES_REQUESTED** [review]

File: cmd/logger.go (hunk 1 above)
Comment (kannappanr) [reply cmd/logger.go:283]: No. Only for startup messages.

File: cmd/signals.go (hunk 4 above)
Comment (kannappanr) [reply cmd/signals.go]: done

File: cmd/prepare-storage.go (hunk 3 above)
Comment (kannappanr) [reply cmd/prepare-storage.go]: done

File: cmd/prepare-storage.go (hunk 2 above)
Comment (kannappanr) [reply cmd/prepare-storage.go]: done

**Review by ebozduman — APPROVED** [review]

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Comment by kannappanr** [comment]
blocked on https://github.com/minio/minio/pull/5678

**Comment by kannappanr** [comment]
Closing this PR. Not needed anymore.

**Pull request closed without merging**